From 25bf7213aa6233584a0c5c017d12991c557516b0 Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Fri, 28 Feb 2025 18:20:35 +0800
Subject: [PATCH] build: Find regex component of Boost explicitly

'Boost::regex' could not be found unless regex is specified explicitly
on some systems.

> Target "libimhex" links to: Boost::regex but the target was not found.

cf. https://github.com/AOSC-Dev/aosc-os-abbs/pull/9798

Co-authored-by: Henry Chen <chenx97@aosc.io>
Signed-off-by: xtex <xtex@aosc.io>
---
 cmake/build_helpers.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/build_helpers.cmake b/cmake/build_helpers.cmake
index 39244d54836d..9342449c5a5a 100644
--- a/cmake/build_helpers.cmake
+++ b/cmake/build_helpers.cmake
@@ -826,7 +826,7 @@ macro(addBundledLibraries)
     endif()
 
     if (USE_SYSTEM_BOOST)
-        find_package(Boost REQUIRED)
+        find_package(Boost REQUIRED CONFIG COMPONENTS regex)
         set(BOOST_LIBRARIES Boost::regex)
     else()
         add_subdirectory(${THIRD_PARTY_LIBS_FOLDER}/boost ${CMAKE_CURRENT_BINARY_DIR}/boost EXCLUDE_FROM_ALL)

base-commit: b6b24dfd291c0f82623b27f00d902ffa4f3a573d
-- 
2.48.1

